src/lib/y2partitioner/widgets/device_resize_button.rb

```diff
@@ -35,13 +38,39 @@ def label
 
     private
 
+      # @see DeviceButton#actions
+      def actions
+        return nil unless device_validation
+        super
+      end
+
+      # Checks whether the device supports resizing
+      #
+      # @note An error popup is shown when the device does not support resizing.
+      #
+      # @return [Boolean] true if the device supports resizing; false otherwise.
+      def device_validation
+        return true if device.is?(:partition, :md)
```

Comment:
Shouldn't we be checking here for software defined MDs instead MDs in general?

Reply:
Never mind, I just realize it's not necessary since that button will never be displayed for BIOS defined RAIDs.

Reply:
Which makes me think. If resizing a partition (actually defining a new size for it) is a completely different action from resizing an MD or an VG (in fact, we don't define a new size but we add/remove devices to it). Why do we share the button class? Just because they have the same label?

Reply:
Well, as commented face to face, the only one reason is to avoid to have 5 identical classes with only 1 different line.